typescript/sdk/src/router/HyperlaneRouterDeployer.ts

```diff
@@ -17,6 +22,54 @@ export abstract class HyperlaneRouterDeployer<
 > extends HyperlaneDeployer<Config, Factories> {
   abstract router(contracts: HyperlaneContracts<Factories>): Router;
 
+  // The ISM check does not appropriately handle ISMs that have sender,
+  // recipient, or body-specific logic. Folks that wish to deploy using
+  // such ISMs *may* need to override checkConfig to disable this check.
+  async checkConfig(configMap: ChainMap<Config>): Promise<void> {
```

Comment:
Feeling a bit  that the semi-complex validation at this layer further blurs the division of responsibility btwn these classes and stuff in hyperlane-deploy. That said, certainly still a net good thing to have.

Reply:
Yeah the boundaries aren't as clear as I'd like, I don't have a good answer.